docs/search/iterative.md

```diff
@@ -1,29 +1,29 @@
-## 简介
+## Introduction
 
-迭代加深是一种 **每次限制搜索深度的** 深度优先搜索。
+Iterative deepening search is a depth-first search that **restricts the search depth each time**. Its abbreviation is IDS or IDDFS.
 
-它的本质还是深度优先搜索，只不过在搜索的同时带上了一个深度 $d$ ，当 $d$ 达到设定的深度时就返回，一般用于找最优解。如果一次搜索没有找到合法的解，就让设定的深度 $+1$ ，重新从根开始。
+The principle of IDS is still depth-first search, but a depth $d$ is also maintained during the search, and returned when $d$ reaches the set depth, which is generally used to find the optimal solution. If a search fails to find a legal solution, let the set depth $+1$ and start from the root again.
 
-既然是为了找最优解，为什么不用 BFS 呢？我们知道 BFS 的基础是一个队列，队列的空间复杂度很大，当状态比较多或者单个状态比较大时，使用队列的 BFS 就显出了劣势。事实上，迭代加深就类似于用 DFS 方式实现的 BFS，它的空间复杂度相对较小。
+Since the goal is to find the optimal solution, why not use BFS? We know that the basis of BFS is a queue, and the space complexity of the queue is very large. When there are many states or a single state is relatively large, BFS with queue is an disadvantage. In fact, iterative deepening search is like a BFS implemented by DFS, and its space complexity is relatively small.
```

Comment:
```suggestion
Since the goal is to find the optimal solution, why not use BFS? We know that the basis of BFS is a queue, and the space complexity of queues are very large. When there are multiple states or a single state is relatively large, BFS with queue is disadvantageous. In fact, iterative deepening search is like a BFS implemented by DFS, and its space complexity is relatively small.
```